addons/account/models/account_move.py

```diff
@@ -1761,24 +1761,25 @@ def _onchange_partner_id(self):
 
         warning = {}
         if self.partner_id:
-            rec_account = self.partner_id.property_account_receivable_id
-            pay_account = self.partner_id.property_account_payable_id
+            # We need to use sudo, since we might not have access to an ancestor company.
```

Comment:
We are not accessing anything on the company here, only it's id to fetch the properties... what am I missing?

Reply:
Good question. On any of these partner accesses I got the following error message, which was solved by using sudo for them.

![Odoo_-_New](https://github.com/odoo/odoo/assets/5985577/d6a6360d-c6b9-4d70-a196-af6d4d9d1451)

After further checking, I found out that the cause of the problem was the line at the top of this function doing
```py
self = self.with_company(self.journal_id.company_id)
```
When I remove that line, it seems to work fine. But I guess it is there for a reason ...

Reply:
> When I remove that line, it seems to work fine. But I guess it is there for a reason ...

Indeed, it is for when you create an invoice for a company that is not selected as the main one in the company selector.

One solution could be to use `self.journal_id.company_id._accessible_branch()`
Still need to work out the way company properties work with the branches :sweat_smile: